main.go

```diff
@@ -0,0 +1,543 @@
+	}
+
+	// at this point we know there is precisely one QueueMessage (== task)
+	qm := queueMessagesList.QueueMessages[0]
+
+	// To find the task referenced in a message the worker must base64
+	// decode and JSON parse the contents of the <MessageText> tag. This
+	// would return an object on the form: {taskId, runId}.
+	m, err := base64.StdEncoding.DecodeString(qm.MessageText)
+	if err != nil {
+		return nil, err
+	}
+
+	// Utility method for replacing a placeholder within a uri with
+	// a string value which first must be uri encoded...
+	detokeniseUri := func(uri, placeholder, rawValue string) string {
+		return strings.Replace(uri, placeholder, strings.Replace(url.QueryEscape(rawValue), "+", "%20", -1), -1)
+	}
+
+	// Before using the signedDeleteUrl the worker must replace the placeholder
+	// {{messageId}} with the contents of the <MessageId> tag. It is also
+	// necessary to replace the placeholder {{popReceipt}} with the URI encoded
+	// contents of the <PopReceipt> tag.  Notice, that the worker must URI
+	// encode the contents of <PopReceipt> before substituting into the
+	// signedDeleteUrl. Otherwise, the worker will experience intermittent
+	// failures.
+
+	// Since urlPair is a value, not a pointer, we can update this copy which
+	// is associated only with this particular task
+	urlPair.SignedDeleteUrl = detokeniseUri(
+		detokeniseUri(
+			urlPair.SignedDeleteUrl,
+			"{{messageId}}",
+			qm.MessageId,
+		),
+		"{{popReceipt}}",
+		qm.PopReceipt,
+	)
+
+	// initialise fields of TaskRun not contained in json string m
+	taskRun := TaskRun{
+		QueueMessage:  qm,
+		SignedURLPair: urlPair,
+	}
+	// now populate remaining json fields of TaskRun from json string m
+	err = json.Unmarshal(m, &taskRun)
+	if err != nil {
+		return nil, err
+	}
+
+	// Workers should read the value of the `<DequeueCount>` and log messages
+	// that alert the operator if a message has been dequeued a significant
+	// number of times, for example more than 10.
+	if qm.DequeueCount > 10 {
+		log.Printf("WARN: Task with id %v has been dequeued %v times!", taskRun.TaskId, qm.DequeueCount)
+	}
+
+	return &taskRun, nil
+}
+
+// Claims the given task
+func (task *TaskRun) claim() error {
+
+	// create payload for API call for claiming task
+	task.TaskClaimRequest = queue.TaskClaimRequest{
+		WorkerGroup: os.Getenv("WORKER_GROUP"),
+		WorkerId:    os.Getenv("WORKER_ID"),
+	}
+
+	// Using the taskId and runId from the <MessageText> tag, the worker
+	// must call queue.claimTask().
+	tcrsp, callSummary := Queue.ClaimTask(task.TaskId, fmt.Sprintf("%d", task.RunId), &task.TaskClaimRequest)
+	task.ClaimCallSummary = *callSummary
+
+	// check if an error occurred...
+	if callSummary.Error != nil {
+		// If the queue.claimTask() operation fails with a 4xx error, the
+		// worker must delete the messages from the Azure queue.
+		if callSummary.HttpResponse.StatusCode/100 == 4 {
```

Comment:
Don't delete if you have a 401 authentication error :)

Generally, if it's not a 404 or 409 error you should probably log the error... but still delete just for good measure :)